src/api/credrank.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+import {CredGraph} from "../core/credrank/credGraph";
+import {
+  type WeightedGraph,
+  merge,
+  overrideWeights,
+} from "../core/weightedGraph";
+import {
+  ensureIdentityExists,
+  toBonusPolicy,
+  type DependenciesConfig,
+} from "../api/dependenciesConfig";
+import {Ledger} from "../core/ledger/ledger";
+import {computeBonusMinting, createBonusGraph} from "../core/bonusMinting";
+import {applyBudget, type Budget} from "../core/mintBudget";
+import {type WeightsT} from "../core/weights";
+import {contractions as identityContractions} from "../core/identity";
+import {credrank as computeCredrank} from "../core/credrank/compute";
+
+export type CredrankInput = {|
+  pluginGraphs: $ReadOnlyArray<WeightedGraph>,
+  ledger: Ledger,
+  dependencies: DependenciesConfig,
+  weightOverrides: WeightsT,
+  pluginsBudget: Budget | null,
+|};
```

Comment:
Is there a reason that these fields are mutable rather than read-only
(e.g., `+pluginGraphs: ...`)? We use read-only struct fields by default,
and I don’t see a need for these to be mutable.

Reply:
accepted